packages/schema/bind/src/bindings/plugin-ts/createPluginContext.ts

```diff
@@ -0,0 +1,69 @@
+import { toImportablePath } from "../../utils/toImportablePath";
+
+import {
+  generateName,
+  InvokableModules,
+  Manifest,
+  MetaManifest,
+} from "@web3api/core-js";
+import { TypeInfo } from "@web3api/schema-parse";
+import { camelCase, upperFirst } from "lodash";
+
+export type PluginContextOptions = {
+  manifest: Manifest;
+  metaManifest?: MetaManifest;
+  typeInfo: TypeInfo;
+};
+
+export type PluginContextModules = {
+  [module in InvokableModules]?: {
+    env?: {
+      modulePath: string;
+    };
+    modulePath: string;
+  };
+};
+
+export type PluginContext = {
+  className: string;
+  funcName: string;
+  modules: PluginContextModules;
+};
+
+export function createPluginContext(opts: PluginContextOptions): PluginContext {
+  const funcName = camelCase(
+    !opts.metaManifest ? generateName() : opts.metaManifest.name
```

Comment:
I think it would work just a side note that if he creates 2 plugins without meta-manifest and want to test it using client he need to use alias or something while importing this plugins. Not a big deal though and having meta-manifest while building plugins should always recommended anyways.

Reply:
True, and we could even move the name to the main manifest, I think I saw a suggestion like that somewhere

Reply:
I am not so sure on this, because I think we can normalise all API type (WASM and Plugin) Manifest to share same required configs and we can have a single CodeGenerator for both Wasm and Plugin, since both have so much similarity after we added support for modules in plugin.